fhir-audit/src/main/java/com/ibm/fhir/audit/logging/api/configuration/ConfigurationTranslator.java

```diff
@@ -0,0 +1,238 @@
+                apiKey = credentials.getApiKey();
+            }
+        }
+
+        // If fails to get config from environment, then try to get them from FHIR config
+        if (bootstrapServers == null || bootstrapServers.length() < 10 || apiKey == null || apiKey.length() < 10) {
+            Objects.requireNonNull(auditLogProperties, "Audit log properties cannot be null.");
+            logger.info("Using FHIR config to find credentials.");
+            bootstrapServers = auditLogProperties.getStringProperty(PROPERTY_AUDIT_KAFKA_BOOTSTRAPSERVERS);
+            apiKey = auditLogProperties.getStringProperty(PROPERTY_AUDIT_KAFKA_APIKEY);
+        }
+
+        // If still fails to get config for kafka producer, then throw <pre>exception</pre>
+        if (bootstrapServers == null || bootstrapServers.length() < 10 || apiKey == null || apiKey.length() < 10) {
+            throw new FHIRException("Can not get kafka settings!");
+        }
+
+        // We load the configuration.
+        checkAndLoadDefaults(props, auditLogProperties);
+
+        // We override if we need to
+        Object o =
+                props.put("sasl.jaas.config", String.format("org.apache.kafka.common.security.plain.PlainLoginModule required username=\"%s\" password=\"%s\";", KAFKA_USERNAME, apiKey));
+        if (o != null) {
+            logger.warning("Environmental properties are overriding fhir-server-config.json for - sasl.jaas.config");
+        }
+        o = props.put("bootstrap.servers", bootstrapServers);
+        if (o != null) {
+            logger.warning("Environmental properties are overriding fhir-server-config.json for - bootstrap.servers");
+        }
+    }
+
+    /*
+     * check and load the default serialiers (for example)
+     */
+    private void checkAndLoadDefaults(Properties props, PropertyGroup auditLogProperties) {
+        if(!props.containsKey("key.serializer")) {
+            props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+        }
+        if(!props.containsKey("value.serializer")) {
+            props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+        }
+        if(!props.containsKey("sasl.mechanism")) {
+            props.put("sasl.mechanism", "PLAIN");
+        }
+        if(!props.containsKey("security.protocol")) {
+            props.put("security.protocol", "SASL_SSL");
+        }
+        if(!props.containsKey("ssl.protocol")) {
+            props.put("ssl.protocol", "TLSv1.2");
+        }
+        if(!props.containsKey("ssl.enabled.protocols")) {
+            props.put("ssl.enabled.protocols", "TLSv1.2");
+        }
+        if(!props.containsKey("ssl.endpoint.identification.algorithm")) {
+            props.put("ssl.endpoint.identification.algorithm", "HTTPS");
+        }
+    }
+
+    /**
+     * gets the location to load the properties from:
+     * 1 - Environment Bindings with EventStreams Credentials format - "env"
```

Comment:
make a list if you want formatting preserved

Reply:
fixed